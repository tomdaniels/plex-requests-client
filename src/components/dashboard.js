import React from 'react';

const Dashboard = () => (
    <div className="content-container">
        Replace me with an app!
    </div>
);

export default Dashboard;