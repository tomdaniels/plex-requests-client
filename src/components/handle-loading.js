import React from 'react';

const Loading = () => (
  <div>
    <img src="/images/loader.gif"/>
  </div>
);

export default Loading;
