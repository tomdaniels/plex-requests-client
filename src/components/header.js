import React from 'react';
import PropTypes from 'prop-types';

const Header = ({title}) => (
    <header>
      <div className="plex-requests__header-wrap">
        <div className="plex-requests__header-left">
          <img
              className="plex-requests__header-img"
              src="/images/play-symbol.png"
              alt="Plex Requests for tomd.io" />
          <p className="plex-requests__header-item">{title}</p>
        </div>
      </div>
    </header>
);

Header.propTypes = {
  title: PropTypes.string.isRequired,
};

export default Header;
