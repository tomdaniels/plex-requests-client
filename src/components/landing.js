import React from 'react';
import Header from './header';
import SearchPage from './search-page';

const Landing = () => (
  <div>
    <Header title="tomd.io | Requests" />
    <SearchPage apiKey="9c5d6b4947c4158889089d104d6ad8b8" />
  </div>
);

export default Landing;
