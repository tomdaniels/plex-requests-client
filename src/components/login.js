import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="login-page__box-layout">
        <div className="login-page__box">
            <h1 className="login-page__title">Boiler plate</h1>
            <p>Insert tagline</p>
            <button className="button" onClick={startLogin}>Login with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);