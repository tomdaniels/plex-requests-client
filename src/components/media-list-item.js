import React from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import Modal from 'react-modal';
import Seasons from './seasons-list';

class MediaListItem extends React.Component {
  state = {
    isLoading: false,
    expandTvShow: false,
    sucessful: false,
    error: false,
    seasons: [],
  };

  toggleModal = () => {
    this.setState((prevState) => ({
      expandTvShow: !prevState.expandTvShow,
      sucessful: false,
      error: false,
    }));
  };

  onMovieRequestClick = () => {
    this.setState(() => ({
      isLoading: true,
    }));
    const endpoint = `http://requests-api.tomd.io/v1/movie/${this.props.id}`;
    axios.post(endpoint).then((response) => {
      if (response.status === 200) {
        this.setState(() => ({
          isLoading: false,
          successful: true,
        }));
        alert(`${this.props.title} has successfully been requested`);
      }
    }).catch((error) => {
      this.setState(() => ({
        isLoading: false,
        error: true,
      }));
      alert(`Oops... Call the tech guys! Something went wrong requesting ${this.props.title}`);
    });
  };

  onFullSeriesClick = () => {
    this.setState(() => ({
      isLoading: true,
    }));
    const endpoint = `http://requests-api.tomd.io/v1/tv/${this.props.id}`;
    axios.post(endpoint).then((response) => {
      if (response.status === 200) {
        this.setState(() => ({
          isLoading: false,
          successful: true,
        }));
        alert(`${this.props.title} has successfully been requested`);
      }
    }).catch((error) => {
      this.setState(() => ({
        isLoading: false,
        error: true,
      }));
      alert(`Oops... Call the tech guys! Something went wrong requesting ${this.props.title}`);
    });
  };

  getSeasonData = () => {
    const endpoint = `https://api.themoviedb.org/3/tv/${this.props.id}?api_key=${this.props.apiKey}`;

    axios.get(endpoint).then((response) => {
      if (response.status === 200) {
        const results = response.data.seasons;
        const seasons = results.map((result) => ({
          id: result.id,
          date: result.air_date,
          epCount: result.episode_count,
          name: result.name,
          seriesId: this.props.id,
        }));
        this.setState(() => ({
          seasons,
        }));
      }
    }).catch((err) => console.log(err));
  };

  render() {
    return (
      <div>
        <div className="media-list__title">
          <h5 className="media-list__media-name">{this.props.title}</h5>
          <h5 className="media-list__media-date">{this.props.date}</h5>
        </div>
        <div className="media-list__description">
          {
            this.props.imageSlug !== 'null' &&
              <img className="media-list__poster"
              src={`http://image.tmdb.org/t/p/w185${this.props.imageSlug}`}
              alt={`${this.props.source === 'tv' ? 'Series' : 'Movie'} Poster, ${this.props.title}`}
              />
          }
          <p>
            {
              this.props.desc || `Sorry, there is no description available for this
              ${this.props.source === 'tv' ? 'series' : 'movie'}. `
            }
          </p>
        </div>
        <div className="media-list__button-wrap">
          {
            this.props.source === 'tv' ? (
              <div>
                <button
                  className="media-list__button"
                  onClick={this.onFullSeriesClick}
                >
                {this.state.isLoading ? (
                  <img class="media-list__loader" src="/images/loader.gif"/>
                ) : 'Request Entire Series'}
                </button>
                <button
                  className="media-list__button"
                  onClick={this.toggleModal}
                >
                  +
                </button>
              </div>
            ) : (
              <button
                className="media-list__button"
                onClick={this.onMovieRequestClick}
              >
                {this.state.isLoading ? (
                  <img class="media-list__loader" src="/images/loader.gif"/>
                ) : 'Request Movie'}
              </button>
            )
          }
        </div>
        {
          this.state.expandTvShow &&
            <Modal
              isOpen={this.state.expandTvShow}
              onAfterOpen={this.getSeasonData}
              onRequestClose={this.toggleModal}
              ariaHideApp={false}
              className="modal"
            >
              <Seasons title={this.props.title} seasons={this.state.seasons} />
            </Modal>
        }
      </div>
    )
  }
};



MediaListItem.propTypes = {
  apiKey: PropTypes.string.isRequired,
  media: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    desc: PropTypes.string,
    date: PropTypes.number,
    image: PropTypes.string,
  }),
};

export default MediaListItem;
