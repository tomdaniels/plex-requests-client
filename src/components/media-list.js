import React from 'react';
import PropTypes from 'prop-types';
import faker from 'faker';
import remove from 'lodash/remove';
import ScrollAppear from './scroll-appear';
import MediaListItem from './media-list-item';

const MediaList = ({ media, apiKey }) => {
  const fitleredMedia = (
    media
      .filter((item, index, self) => (
        index === self.findIndex((t) => (
          t.title === item.title && t.id === item.id
        ))
      ))
      .sort((a, b) => b.popularity - a.popularity)
    );

  return (
    <div>
      <h4 className="media-list__section-header">
        {fitleredMedia.length === 1 ? `1 result found` : `${fitleredMedia.length} results found`}
      </h4>
      <ul>
        {
          fitleredMedia.length > 0 && fitleredMedia.map((mediaItem) => (
            <li key={mediaItem.id} className="plex-requests__media-list">
              <ScrollAppear
                direction={faker.random.arrayElement(["left","right"])}
                children={ <MediaListItem apiKey={apiKey} {...mediaItem} /> }
              >
              </ScrollAppear>
            </li>
          ))
        }
      </ul>
    </div>
  );
}

MediaList.propTypes = {
  apiKey: PropTypes.string.isRequired,
  media: PropTypes.arrayOf(PropTypes.objectOf(
    PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number
      ])))
};

MediaList.defaultProps = {
  media: [],
};

export default MediaList;
