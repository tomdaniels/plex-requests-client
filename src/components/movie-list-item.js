import React from 'react';
import PropTypes from 'prop-types';

const MovieListItem = ({ id, title, desc, date, image }) => (
  <div>
    <div>
      <h5>{title}</h5>
      <h5>{date}</h5>
    </div>
    <div>
      <p>{desc}</p>
    </div>
    <div>
      <button>Request</button>
    </div>
  </div>
);

MovieListItem.propTypes = {
  movie: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    desc: PropTypes.string,
    date: PropTypes.number,
    image: PropTypes.string,
  }),
};

export default MovieListItem;
