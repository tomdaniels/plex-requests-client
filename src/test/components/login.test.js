import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/login';

test('should render the login page correctly', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should call start login on button click', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin} />);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});