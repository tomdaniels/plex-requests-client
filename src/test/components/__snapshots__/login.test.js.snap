// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the login page correctly 1`] = `
<div
  className="login-page__box-layout"
>
  <div
    className="login-page__box"
  >
    <h1
      className="login-page__title"
    >
      Boiler plate
    </h1>
    <p>
      Insert tagline
    </p>
    <button
      className="button"
    >
      Login with Google
    </button>
  </div>
</div>
`;
